step1-06/demo/README.md

```diff
@@ -61,53 +61,55 @@ return (
+- `id`: The `key` is not actually passed into the component, so we pass the same value as `id` as well. This will help us out later.
+- `todos[id]`: Lastly we use the `id` to grab the todo from our `todos` object, then use the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) to pass through the todo's `label` and `completed` values.
   > This spread operator is the same as saying `label={todos[id].label} completed={todos[id].completed}`. Pretty obvious why spread is so handy!
 
-### State Driven and Stateful Header
+### State-Driven and Stateful Header
 
 Within the header we've got a situation where we not only want to pass `filter` state down to it, but we also want to maintain state within the control. Fortunately, this is no problem at all for React. First off let's deal with the incoming state.
 
-#### Conditional ClassNames
+#### Conditional Class Names
 
-In CSS based styling, visual states are applied by adding and removing classes. We can use the filter value to conditionally add a class thereby lighting up the correct filter button.
+In CSS-based styling, visual states are applied by adding and removing classes. We can use the filter value to conditionally add a class, thereby lighting up the correct filter button.
 
 ```jsx
 <nav className="filter">
-  <button className={filter == 'all' ? 'selected' : ''}>all</button>
-  <button className={filter == 'active' ? 'selected' : ''}>active</button>
-  <button className={filter == 'completed' ? 'selected' : ''}>completed</button>
+  <button className={filter === 'all' ? 'selected' : ''}>all</button>
+  <button className={filter === 'active' ? 'selected' : ''}>active</button>
+  <button className={filter === 'completed' ? 'selected' : ''}>completed</button>
 </nav>
 ```
 
-> Ternary operators are very popular in React code as each expression could be a string for a className, or even a JSX element.
+> The [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) `condition ? expressionIfTrue : expressionIfFalse` is widely used in React code, as each expression could be a string for a className or even a JSX element.
 
-#### Creating a Controlled Input
+#### Adding a Controlled Input
 
-In tradition HTML forms users interact with the form, and on submit, those values are captured and transmitted. Those are called **uncontrolled inputs**. A **controlled input** is one whos value is defined by state, and interaction with that input updates state with each keystroke. This round trip process might sound inefficient, but in reality it has little to no impact, and it enables some advanced form functionality.
+In traditional HTML forms, users interact with the form, and on submit, those values are captured and transmitted. Those are called **uncontrolled inputs**.
```

Comment:
I'd like to refine this part some more since this is a *very* significant source of confusion, but I'll keep working on the wording and make a separate PR.